src/js/controller.js

```diff
@@ -234,6 +254,97 @@ export default class Controller extends Component {
             && this.state.subindicator.indicatorId === filteredIndicator.indicatorId) {
             this.triggerEvent('mapchip.choropleth.filtersUpdated', filteredIndicator);  // mapchip will be updated
         }
+        this.updateShareUrl();
+    }
+
+    updateShareUrl(){
+      let selectedFilters = [];
+      this._filteredIndicators.map(
+        (indicatorFilter) => {
+          const nonDefaultFilters = indicatorFilter.filters.filter(f => f.isDefault !== true);
+          if (nonDefaultFilters.length > 0){
+            selectedFilters.push({
+              "indicatorId": indicatorFilter.indicatorId,
+              "filters": nonDefaultFilters
+            })
+          }
+        }
+      )
+      let currentState = {"filters": selectedFilters}
+
+      const urlParams = new URLSearchParams(window.location.search);
+      const profileView = JSON.parse(urlParams.get("profileView"));
+      if (selectedFilters.length > 0){
+        if (profileView === null){
+          history.pushState(
+            {
+              "filters": this._filteredIndicators,
+              "profileView": currentState
+            },
+            '',
+            `?profileView=${encodeURIComponent(JSON.stringify(currentState))}${window.location.hash}`
+          );
+        } else {
+          if(!isEqual(profileView, currentState)){
+            history.pushState(
+              {
+                "filters": this._filteredIndicators,
+                "profileView": currentState
+              },
+              '',
+              `?profileView=${encodeURIComponent(JSON.stringify(currentState))}${window.location.hash}`
+            );
+          }
+        }
+      } else if (selectedFilters.length === 0 && profileView !== null) {
+        history.pushState(
+          {
+            "filters": this._filteredIndicators,
+            "profileView": currentState
+          }, '', `/${window.location.hash}`
+        );
+      }
```

Comment:
It looks like the first two pushState are identical and the third pushState is very similar. That makes be think some of these if statements are not needed or there can be variable to help reduce the repeated code like

```
let stateParams = {...}

if (...) {
  pushState(stateParams, "location1");
}
else {
  pushState(stateParams, "location2");
}
```